prow/github/client.go

```diff
@@ -229,8 +230,22 @@ func (c *Client) doRequest(method, path, accept string, body interface{}) (*http
 	return c.client.Do(req)
 }
 
-func (c *Client) BotName() string {
-	return c.botName
+func (c *Client) BotName() (string, error) {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+	if c.botName == "" {
+		var u User
+		_, err := c.request(&request{
+			method:    http.MethodGet,
+			path:      fmt.Sprintf("%s/user", c.base),
+			exitCodes: []int{200},
+		}, &u)
+		if err != nil {
+			return "", err
```

Comment:
nit: You should return a new error that contains `err`'s message but also notes that there was a failure getting the bot name instead of doing that in `report.go`.